src/quacc/recipes/onetep/_base.py

```diff
@@ -0,0 +1,67 @@
+"""Base jobs for Onetep."""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from ase.calculators.onetep import Onetep, OnetepProfile
+
+from quacc import SETTINGS
+from quacc.runners.ase import run_calc
+from quacc.schemas.ase import summarize_run
+from quacc.utils.dicts import recursive_dict_merge
+
+if TYPE_CHECKING:
+    from typing import Any
+
+    from ase import Atoms
+
+    from quacc.schemas._aliases.ase import RunSchema
+
+
+def base_fn(
+    atoms: Atoms,
+    calc_defaults: dict[str, Any] | None = None,
+    calc_swaps: dict[str, Any] | None = None,
+    additional_fields: dict[str, Any] | None = None,
+    copy_files: list[str] | None = None,
+) -> RunSchema:
+    """
+    Base function to carry out Onetep recipes.
+
+    Parameters
+    ----------
+    atoms
+        Atoms object
+    calc_defaults
+        The default calculator parameters.
+    calc_swaps
+        Custom kwargs for the ONETEP calculator. Set a value to
+        `quacc.Remove` to remove a pre-existing key entirely. For a list of available
+        keys, refer to the `ase.calculators.onetep.Onetep` calculator.
+    additional_fields
+        Any additional fields to supply to the summarizer.
+    copy_files
+        Files to copy to the runtime directory.
+
+    Returns
+    -------
+    RunSchema
+        Dictionary of results from [quacc.schemas.ase.summarize_run][]
+    """
+    calc_flags = recursive_dict_merge(calc_defaults, calc_swaps)
+
+    atoms.calc = Onetep(
+        input_atoms=atoms,
+        calc_defaults=calc_defaults,
+        pseudo_path=str(SETTINGS.ONETEP_PP_PATH) if SETTINGS.ONETEP_PP_PATH else ".",
+        profile=OnetepProfile(
+            SETTINGS.ONETEP_CMD
+        ),  # TODO: Need to insert SETTINGS.ONETEP_PARLLEL_CMD here.
```

Comment:
Do you have any idea how to appropriately handle what I wrote in the TODO? Namely, right now, the ONETEP command (a binary) is passed, but in practice we would want to allow the user to specify a `ONETEP_PARALLEL_CMD` (e.g. `srun -N 2 blah blah`) that, when combined with the `ONETEP_CMD`, would be the full command being executed. I'm still a bit confused about how this all works with the new `GenericFileIOCalculator` interfaces (see my [ASE issue](https://gitlab.com/ase/ase/-/issues/1415)).

If you have a suggestion on how to incorporate the `ONETEP_PARALLEL_CMD` setting appropriately, that would be great!

Reply:
ah damn I must have missed it. I will address it as soon as I can and I will ping it when I have checked everything again

Reply:
No worries, @Nekkrad. Actually, I just got off a call with the main ASE developer about this particular question. It looks like there is no clean mechanism for specifying commands that prepend the binary. See [here](https://gitlab.com/ase/ase/-/issues/1415). Hopefully, if he agrees with my proposal, then a `prepend_args` command will be included shortly and we can put the parallelization flags there. It sounds like it's very possible the current mechanism, e.g. via `parallel_info`, may be changed or removed. 

So, with that being said, please do what makes the most sense given the current limitations, and I can change things if/when ASE is updated.

Reply:
I could implement something similar to your VASP setup, but I think it is better to stick with parallel info for the time being. This is to allow people already using ASE to easily move to quacc and when/if parallel_info will be removed or if/when your proposal gets accepted it is much easier to change the relative parts of the code

Reply:
Thanks @Nekkrad!

I'll go ahead merge this in with your usage of `parallel_info`, with the expectation that there is a decent likelihood that this may change in the future. Hopefully, at some point in the near future we can move to a more sensible option than a dictionary of parameters, much like the old environment variable approach. We'll see what happens upstream...